### PR TITLE
Rename RedundantAssignEliminator to UnusedAssignEliminator.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libyul/optimiser/Suite.cpp [hunk 1]
```diff
@@ -260,7 +260,7 @@ map<string, char> const& OptimiserSuite::stepNameToAbbreviationMap()
 		{LoadResolver::name,                  'L'},
 		{LoopInvariantCodeMotion::name,       'M'},
 		{ReasoningBasedSimplifier::name,      'R'},
-		{RedundantAssignEliminator::name,     'r'},
+		{UnusedAssignEliminator::name,        'r'},
```
Comment (chriseth) [inline libyul/optimiser/Suite.cpp:263]: I would still keep this at `r`

Comment (cameel) [reply libyul/optimiser/Suite.cpp:263]: Yeah, I think it's better to keep it. Change would require finding it when used in step sequences which is error-prone. Also, `u` is already taken and `U` will probably be taken by the other eliminator...

**Review by cameel — DISMISSED** [review]
I only skimmed over it but looks like it's really only renaming + some docs so as long as it passes tests, it should be fine.

**Comment by Marenz** [comment]
The chromosome seems to have changed, the r position was moved..
```
flcCUnDvejsxIOoighFTLMRmVat_r_pud != 
flcCUnDvejsxIOoighFTLMR_r_mVatpud
```

**Comment by Marenz** [comment]
I pushed a small test update to fix the failing CI

**Pull request merged**